src/main/java/de/tum/in/www1/artemis/domain/VcsRepositoryUrl.java

```diff
@@ -1,58 +1,92 @@
 package de.tum.in.www1.artemis.domain;
 
-import java.net.MalformedURLException;
-import java.net.URL;
+import java.net.URI;
+import java.net.URISyntaxException;
 import java.util.Objects;
 
+// TODO: we might want to rename this class to VcsRepositoryUri
 public class VcsRepositoryUrl {
```

Comment:
It is related because of the refactoring of its use in the communication between VCS and CIS and the additional tests related to SSH URIs which would otherwise fail.